File: Source/JapaneseLanguageTools.Contracts/Models/CharacterGroup.Model.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JapaneseLanguageTools.Contracts.Models;

public class CharacterGroupModel
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [StringLength(256)]
    public string Caption { get; set; } = string.Empty;

    [StringLength(2048)]
    public string? Comment { get; set; }

    public bool Enabled { get; set; }

    public bool AlwaysUse { get; set; }

    public bool Hidden { get; set; }

    [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
    public DateTimeOffset CreatedOn { get; set; }

    [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
    public DateTimeOffset UpdatedOn { get; set; }

    public IList<CharacterModel> Characters { get; set; } = [];
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/CharacterGroup.Model.ts
import CharacterModel from "@/models/Character.Model";

export default interface CharacterGroupModel {
  id: number;
  caption: string;
  comment?: string;
  enabled: boolean;
  alwaysUse: boolean;
  hidden: boolean;
  createdOn?: string;
  updatedOn?: string;

  characters: CharacterModel[];
}
